doc/Gemm.md

```diff
@@ -0,0 +1,405 @@
+# SYCL-BLAS GEMM Developer Documentation
+
+The following is documentation for the `GEMM` kernels and associated areas of code within `SYCL-BLAS` . This should give you a good understanding of how everything works and where/how to do things such as:
```

Comment:
```suggestion
The following is documentation for the `GEMM` kernels and associated areas of code within `SYCL-BLAS`.
This should give you a good understanding of how everything works and where/how to do things such as:
```

I suggest splitting up lines in the whole document. It doesn't have to fit 80 columns, but it should be something like one sentence per line or even more split.